model-optimizer/extensions/front/tf/while_ext.py

```diff
@@ -0,0 +1,169 @@
+"""
+ Copyright (C) 2018-2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+import copy
+
+from extensions.ops.loop import Loop
+from extensions.ops.parameter import Parameter
+from mo.front.extractor import FrontExtractorOp
+from mo.graph.graph import Graph, Node, add_opoutput
+from extensions.front.onnx.loop_ext import connect_body_input, connect_body_output
+from mo.front.extractor import extract_node_attrs
+from mo.front.common.register_custom_ops import check_for_duplicates
+from mo.front.tf.extractor import tf_op_extractor, tf_op_extractors
+from mo.front.tf.extractors.utils import tf_dtype_extractor
+from mo.ops.op import PermuteAttrs
+
+
+def extend_body_graph(body_graph: Graph, body_graph_proto, body_parameter_names, body_results, prefix_name='body/'):
+    for pb_node in body_graph_proto['node_def']:
+        # create an NX node
+        id = prefix_name + body_graph.unique_id(pb_node.name)
+        body_graph.add_node(id, pb=pb_node, kind='op')
+
+        # add incoming edges based on data_nodes_map
+        for dst_port, inp in enumerate(pb_node.input):
+            src_id = inp.split(":")[0]
+            if src_id not in body_parameter_names:
+                src_id = prefix_name + src_id
```

Comment:
Why do you add prefix to body nodes and not for the body Parameters nodes? Can we make the behaviour consistent?

Reply:
I've removed it at all and use `unique_id` now.